src/Dao/CategoryDao.php

```diff
@@ -72,7 +81,7 @@ public function getCategoriesByParentId(string $parentid, int $languageId, int $
         $result = $queryBuilder->execute();
 
         if (!$result instanceof \Doctrine\DBAL\Driver\Statement) {
-            return $categories;
+            throw new CategoryNotFoundException('Error accessing database.');
```

Comment:
I think this is more expressive:
```suggestion
            throw new \UnexpectedValueException("Category-Query result must be instance of \Doctrine\DBAL\Driver\Statement");
```
To the client this will be a internal server error (and he should not know about this problem details anyway), but it will be logged in our log file.